### PR TITLE
Testable with Improvements

### PR DESCRIPTION
**Pull request by LuckyNoS7evin**
#15 

- Added Test Library
- Added IHttpCallHandler interface
- Took old WebRequest and new HttpClient into possible HttpCallHandlers
- Means we can now write tests

#16 

- No Initialize method
- Set Credentials through Settings
- Check Credentials through Settings
- No need to skip validation as it's only done if dev asks

### CODE REVIEW

File: TwitchLib.Api/ApiSettings.cs [hunk 1]
```diff
@@ -84,12 +68,13 @@ public void DynamicScopeValidation(AuthScopes requiredScope, string accessToken
...
         #endregion
+
         #region ValidAccessToken
-        private async Task<bool> ValidAccessTokenAsync(string accessToken)
+        private bool ValidAccessTokenAsync(string accessToken)
         {
             try
             {
-                var resp = await _api.Root.v5.GetRootAsync(accessToken);
+                var resp =  _api.Root.v5.GetRootAsync(accessToken).GetAwaiter().GetResult();
```
Comment (Syzuna) [inline TwitchLib.Api/ApiSettings.cs]: Not sure but this might break UI Applications again? Needs testing

File: TwitchLib.Api/ApiSettings.cs [hunk 2]
```diff
@@ -36,46 +36,30 @@ public void ValidateScope(AuthScopes requiredScope, string accessToken = null)
...
-        #endregion
 
         #region ValidClientId
-        private async Task<bool> ValidClientIdAsync(string clientId)
+        private bool ValidClientIdAsync(string clientId)
         {
             try
             {
-                var result = await _api.Root.v5.GetRootAsync(null, clientId);
+                var result = _api.Root.v5.GetRootAsync(null, clientId).GetAwaiter().GetResult();
```
Comment (Syzuna) [inline TwitchLib.Api/ApiSettings.cs]: Not sure but this might break UI Applications again? Needs testing

**Comment by Mahsaap** [comment]
As suggested in Discord, Was thinking that CheckCredentials might be good to have a confirmation string. Rather then just returning an empty string. Just to make things more transparent to the end user.

**Comment by swiftyspiffy** [comment]
@Mahsaap You thinking like new KeyValuePair<bool, string>(true, "Check successful") or something?

Also perhaps we want to either add documentation to that method defining what the return means to the developer, or maybe just a model that has a bool "Result" and "Response Message" or something.

**Comment by Mahsaap** [comment]
yes something like that

**Comment by LuckyNoS7evin** [comment]
Cool, I think, we'll do that, give me a couple hours and I'll get that in. Just got up need time ;)

**Comment by LuckyNoS7evin** [comment]
@swiftyspiffy done, new result model and method xml included

**Pull request merged**